packages/data-sdk/src/Device.ts

```diff
@@ -304,13 +304,10 @@ export class Device extends EventEmitter implements IRealtimeDevice {
             this.initConnectionMonitoring();
             this.rtcClient = rtcClient;
             this.emit("connect");
-            return;
+            i = 100;
           }
           await delay(100);
         }
-        throw new Error(
-          "A session was created, but the connection could not be established, possibly due to network issues or misconfigured settings."
-        );
```

Comment:
hmmm, i like this rather than silently thinking its okay. I think the `return` is fine in the loop, but can you make this a `console.error` or `console.info` instead of a hard throw?

also, i think method might _"flow"_ if some of these conditions were _reversed_ and the `throw` terminated the method _early_

Reply:
also, the `rtcClient` should probably be _released/shutdown/disposed_ if it _can't_ finish the connection, otherwise its going to linger around _forever_ (Cause its been created)